cmd/HocusPocus.hs

```diff
@@ -64,11 +64,13 @@ app command session = do
     (\conn -> do
         let offer = FileTransfer.Message "Brave new world that has such offers in it"
         Peer.sendMessage conn (toS (Aeson.encode offer)))
+  pure 42
```

Comment:
I'm not sure I follow these `app` changes.  They don't seem necessary to the main focus  of the branch.  My first guess was that `app` needed to be adjusted to follow the type of some other part of the system but that doesn't seem to be the case (`Session -> IO a` being the type the function it is passed to expects).

My other guess is that the new `print result` in `main` is useful and desirable because it's what will allow error reporting to the user (such as a `ConnectionClosed` event).  And that `print ()` (probably not the right syntax) somehow is not cool.  I wasn't able to get the code to run to completion in the success case because I couldn't figure out how to make a `send` and `receive` hook up with each other (seems like there's no wormhole code emitted or accepted by hocus-pocus?) so I wasn't able to test this hypothesis.  I did notice that (a) `hocus-pocus send` and `hocus-pocus receive` both print out "hocus-pocus: ConnectionClosed` when I C-c them and (b) `stack build` succeeds if I revert the changes to `HocusPocus.hs`.

Reply:
Oh, right. I can see why this would be confusing.

These changes function as an ad hoc test showing that the value returned from `app` is returned from `Rendezvous.runClient`. Does that help?

Reply:
* `print ()` works just fine. `()` is a legitimate value of the type `()` (puns!), and indeed the only value of that type. The type `()` implements `Show`, so it can be printed.
* `hocus-pocus` hardcodes a magic wormhole password, `potato`. Joining the nameplate and the password together isn't strictly speaking a feature of the protocol, but rather of the Python client.
* The `hocus-pocus: ConnectionClosed` you see is what you get when an exception is thrown but not caught.

Reply:
Yep, thanks.